src/main/java/org/broadinstitute/hellbender/utils/R/RScriptLibrary.java

```diff
@@ -22,19 +24,27 @@ public String getLibraryName() {
     }
 
     public String getResourcePath() {
-        return name + ".tar.gz";
+        return name + R_LIBRARY_SUFFIX;
     }
 
     /**
      * Writes the library source code to a temporary tar.gz file and returns the path.
-     * @return The path to the library source code. The caller must delete the code when done.
+     * @return The path to the library source code.
      */
     public File writeTemp() {
         return IOUtils.writeTempResourceFromPath(getResourcePath(), RScriptLibrary.class);
     }
 
-    public File writeLibrary(File tempDir) {
-        File libraryFile = new File(tempDir, getLibraryName());
+    /**
+     * Retrieve the resource for this library and write it to a File in {@code targetDir}. The File is not
+     * automatically scheduled for deletion and must be cleaned up by the caller.
+     * @param targetDir target directory where the File should be written
+     * @return the newly created File containing the library
+     */
+    public File writeLibrary(final File targetDir) {
```

Comment:
Can you have this method register its libraryFile for deletion on exit using createTempFile(), rename the method and adjust the javadoc appropriately? You'll need to add an overload of createTempFile() that takes a directory argument.